libs/NfcCoreLib/lib/LibNfc/phLibNfc_Se.c

```diff
@@ -1329,6 +1383,84 @@ static NFCSTATUS phLibNfc_SetSeModeSeqComplete (void* pContext,NFCSTATUS Status,
     return wStatus;
 }
 
+static NFCSTATUS
+phLibNfc_SePowerAndLinkCtrlSeq(void *pContext,
+                               NFCSTATUS wStatus,
+                               void *pInfo)
+{
+    NFCSTATUS wIntStatus = wStatus;
+    pphLibNfc_LibContext_t pLibContext = (pphLibNfc_LibContext_t)pContext;
+    UNUSED(pInfo);
+    PH_LOG_LIBNFC_FUNC_ENTRY();
+    if ((NULL != pLibContext) && (phLibNfc_GetContext() == pLibContext))
+    {
+        if (NFCSTATUS_SUCCESS == wIntStatus)
+        {
+            wIntStatus = phNciNfc_Nfcee_SePowerAndLinkCtrlSet(pLibContext->sHwReference.pNciHandle,
+                                    (void *)pLibContext->sSeContext.pActiveSeInfo->hSecureElement,
+                                    pLibContext->sSeContext.ePowerLinkMode,
+                                    (pphNciNfc_IfNotificationCb_t)&phLibNfc_InternalSequence,
+                                    (void *)pLibContext);
+        }
+    }
+    else
+    {
+        PH_LOG_LIBNFC_CRIT_STR("Invalid Context passed from lower layer!");
+    }
+    PH_LOG_LIBNFC_FUNC_EXIT();
+    return wIntStatus;
+}
+
+static NFCSTATUS phLibNfc_SePowerAndLinkCtrlEnd(void *pContext, NFCSTATUS wStatus, void *pInfo)
+{
+    pphLibNfc_LibContext_t pLibContext = pContext;
+    UNUSED(pInfo);
+    PH_LOG_LIBNFC_FUNC_ENTRY();
+    if ((NULL != pLibContext) && (phLibNfc_GetContext() == pLibContext))
+    {
+        if (NFCSTATUS_SUCCESS == wStatus)
+        {
+            PH_LOG_LIBNFC_INFO_STR("Set Se Power Mode success");
+        }
+        else
+        {
+            PH_LOG_LIBNFC_CRIT_STR("Set Se Power Mode Failed!");
+        }
+    }
+    PH_LOG_LIBNFC_FUNC_EXIT();
+    return wStatus;
+}
+
+static NFCSTATUS phLibNfc_SePowerAndLinkCtrlCompleteSequence(void* pContext, NFCSTATUS Status, void *pInfo)
+{
+    NFCSTATUS wStatus = Status;
+    pphLibNfc_LibContext_t pCtx = phLibNfc_GetContext();
+    pphLibNfc_RspCb_t ClientCb = NULL;
+    void *ClientContext = NULL;
+    phLibNfc_Handle hSeHandle;
+    UNUSED(pInfo);
+    PH_LOG_LIBNFC_FUNC_ENTRY();
+    if ((NULL != pCtx) && (pContext == pCtx))
+    {
+        ClientCb = pCtx->CBInfo.pPowerCtrlLinkCb;
+        ClientContext = pCtx->CBInfo.pPowerCtrlLinkCntx;
+        hSeHandle = pCtx->sSeContext.pActiveSeInfo->hSecureElement;
+
+        pCtx->StateContext.Flag = phLibNfc_StateTransitionComplete;
```

Comment:
Out of interest, why are you setting 'StateContext.Flag' here?

Reply:
I don't know and I believe it is a mistake.